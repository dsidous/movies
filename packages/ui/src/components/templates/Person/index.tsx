import React from 'react';
import { IntlProvider } from 'react-intl';

import PageTransition from '../../atoms/PageTransition/index';
import PersonProfile from '../../organisms/PersonProfile';
import { Person } from '@typesRoots/person';

interface Props {
  person: Person;
}

const Person: React.FC<Props> = ({ person }) => {
  return (
    <PageTransition>
      <IntlProvider locale={navigator.language}>
        <PersonProfile person={person} />
      </IntlProvider>
    </PageTransition>
  );
};

export default Person;
