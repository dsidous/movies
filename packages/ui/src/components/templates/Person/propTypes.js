import { object, bool } from 'prop-types';

export const propTypes = {
  loading: bool,
  person: object,
};

export const defaultProps = {};
