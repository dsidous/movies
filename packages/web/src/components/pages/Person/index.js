import { compose, withProps } from 'recompose';
import { Person } from '@movies/ui';
import { withPerson } from '@movies/common';

export default compose(
  withProps(props => ({ personId: props.match.params.personId })),
  withPerson(),
)(Person);
