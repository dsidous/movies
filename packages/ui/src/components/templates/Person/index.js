import React from 'react';
import { IntlProvider } from 'react-intl';

import { propTypes } from './propTypes';
import Spinner from '../../atoms/Spinner';
import PageTransition from '../../atoms/PageTransition/index';
import PersonProfile from '../../organisms/PersonProfile';

const Person = props => {
  const { loading } = props;

  if (loading) {
    return <Spinner />;
  }

  const { person } = props;

  return (
    <PageTransition>
      <IntlProvider locale={navigator.language}>
        <PersonProfile person={person} />
      </IntlProvider>
    </PageTransition>
  );
};

Person.propTypes = propTypes;

export default Person;
