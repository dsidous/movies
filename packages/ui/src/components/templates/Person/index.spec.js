import React from 'react';
import { shallow } from 'enzyme';
import Person from '.';

const mockProps = {
  person: {
    id: 1,
  },
  loading: false,
};

describe('Templates/Person', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<Person {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render a spinner when loading expected', () => {
    const wrapper = shallow(<Person {...mockProps} loading />);

    expect(wrapper).toMatchSnapshot();
  });
});
