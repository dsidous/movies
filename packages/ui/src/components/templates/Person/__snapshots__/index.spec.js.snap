// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Templates/Person should render a spinner when loading expected 1`] = `ShallowWrapper {}`;

exports[`Templates/Person should render as expected 1`] = `ShallowWrapper {}`;
